### ISSUE TITLE
[2.4] audio(mic): overconstrainederror

### ISSUE THREAD

**Issue by schrd** [bigbluebutton/bigbluebutton#14429]
**Describe the bug**

Under some rare conditions I cannot join to BBB by microphone. In the browser console I get the following log message:
~~~
sip.SessionDescriptionHandler | SessionDescriptionHandler.getDescription failed - OverconstrainedError: Constraints could be not satisfied.
~~~

Removing microphone permission from the browser does not mitigate this issue.
Once I delete the `BBB_audioInputDeviceId` key in the session storage, the error no longer appears. Firefox then asks again for microphone permission and I can establish an audio connection.

**To Reproduce**

Unfortunately I have not yet found a way to reliably trigger this bug. I use the noisetorch application as a microphone filter. Noisetorch creates a virtual microphone in pulseaudio. This microphone is the one I granted permissions in the browser.

**Expected behavior**
I think if the browser throws a OverconstrainedError BBB should remove the `BBB_audioInputDeviceId` key in the session storage and retry to establish a connection.

**Screenshots**

![Bildschirmfoto von 2022-02-21 22-31-29](https://user-images.githubusercontent.com/179952/155028905-39800b4a-db6c-4055-90d0-27a4016eb25e.png)

**BBB version:**
2.4.4

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox
 - Version 91.6 ESR (however I observed this with other versions as well)

**Smartphone (please complete the following information):**
 - Browser [e.g. stock browser, Safari]

**Comment by schrd** [comment]
I found a way to reliably trigger this bug. You need at least 2 microphones.

* start a meeting, join by microphone
* change the microphone in BBB → device id is stored in Browser session storage
* disconnect audio
* open developer tools in browser, change the value of `BBB_audioInputDeviceId` in the session storage
* reconnect audio, use microphone
* see the error

**Comment by prlanzarin** [comment]
Forgot about this one.
I'll look into it next week - although I may fix it in 2.6 initially (where rolling back from gUM calls should be way easier than <= 2.5) and think about a backport later.

**Comment by prlanzarin** [comment]
I don't intend to backport this to 2.5. Fixed in 2.6, closing as addressed.
